tests/test_cli_compile.py

```diff
@@ -2925,7 +2925,11 @@ def test_pass_pip_cache_to_pip_args(tmpdir, runner, current_resolver):
         cli, ["--cache-dir", str(cache_dir), "--resolver", current_resolver]
     )
     assert out.exit_code == 0
-    assert os.listdir(os.path.join(str(cache_dir), "http"))
+    # TODO: Remove hack once testing only on v23.3+
+    try:
+        assert os.listdir(os.path.join(str(cache_dir), "http"))
+    except FileNotFoundError:
+        assert os.listdir(os.path.join(str(cache_dir), "http-v2"))
```

Comment:
I'd swap the main variant and the fallback so the newer one always comes first.

Reply:
Also, it's more reliable to check the pip version and make a precise assertion. It's not recommended to have any logic in tests.

Reply:
I agree. Any suggestion how to do that?

Reply:
I'd `git grep` the repository for version check examples..

Reply:
I'd use `importlib.metadata`.

Reply:
It's a good idea to follow the same style everywhere, whatever it is.

Reply:
Thanks, didn't read your comment. I found an approach and a corresponding function.

Reply:
Thanks, marking as resolved.